src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -23,7 +23,7 @@ public DirectoryInfo(String path)
         }
 
         [System.Security.SecuritySafeCritical]
-        internal DirectoryInfo(String fullPath, IFileSystemObject fileSystemObject) : base(fileSystemObject)
+        internal DirectoryInfo(String fullPath, bool ignoreThis)
```

Comment:
The ignored parameter is unfortunate. Would be nice if we could refactor this to be a little clearer. Perhaps an Initialize method with bool fullyQualified? Then you also call from the public and new findData constructors?